msteams-platform/concepts/deploy-and-publish/appsource/prepare/frequently-failed-cases.md

```diff
@@ -1,246 +1,348 @@
+* Notification-only bots must send a welcome message that includes a message conveying, "I am a notification-only bot and will not be able to reply to your chats".
 
-* A tab configured by a member in a channel scope must be accessible to the other members without having to seek permissions from the member who configured the tab.
-* The app must provide the permission management options upfront if the tab is for private or restricted use or requires any permissions from the member who configured the tab.
+### 6.3 Bot message spamming
 
-### &#9989; Bots must always be responsive and fail gracefully
+Bots must not spam users by sending multiple messages in short succession.
 
-Your bot must be responsive to any command and not dead-end the user. Here are some tips to help your bot intelligently respond to users:
+* **Bot channel messages**: Don't spam users by creating separate new chat posts. Create a single thread post with replies in the same thread.
+* **Bot personal messages**: Don't send multiple messages. Send one message with complete information. Repeating the same welcome message over regular intervals is not allowed and is considered spamming.
 
-* **Use command lists**. Analyzing user input or predicting user intent is hard. Instead of letting users guess what your bot can do, provide a list of commands your bot understands.
+### 6.4 Bot notifications
 
-![Flow command list](../../../../assets/images/faq/flow-bot.png)
+Bot notifications must include content relevant for the scope you define for the bot (`team`, `groupchat`, or `personal`).
 
-* **Include a help command**. Users are likely to type "Help" when they are lost or when your bot doesn't respond as expected. Include a help command that describes how your app's value will be experienced along with all valid commands.
+### 6.5 Bots and Adaptive Cards
 
-![Flow help command](../../../../assets/images/faq/flow-help.png)
+Adaptive Cards are a common way to display bot messages. Your cards must be lightweight and include only 1-3 actions. If you need to display more content, consider using a task module or tab.
 
-* **Include help content or guidance when your bot is lost**. When your bot cannot understand the user input, it must suggest an alternative action. For example, *"I'm sorry, I don't understand. Type "help" for more information."* Don't respond with an error message or simply, *"I don't understand"*.
+See the following resources for more information:
 
-### &#9989; Help command response
+* [Designing Adaptive Cards](~/task-modules-and-cards/cards/design-effective-cards.md)
+* [Cards reference](~/task-modules-and-cards/cards/cards-reference.md#types-of-cards)
 
```

Comment:
@heath-hamilton: Include tip box for - 

- To improve the user experience, consider including a ‘Help’ Tab along with personal bot that can be used to elaborate more on bot response to help command

- Provide an 'interactive tour' in form of carousel cards or 'try it' buttons to deliver welcome message highlighting the value of the bot and  it's primary functions

Reply:
Just confirming if these are validation guidelines or more "nice to have"? If the latter, do we need these here or on another page?

Reply:
@heath-hamilton, These are nice to have & required for app to score high Quality score. 
We should include quality parameters inside tip box in relevant section. That way user can differentiate validation guidelines with best practice & at the same time find all relevant info at same section